### PR TITLE
[SPARK-6291] [MLLIB] GLM toString & toDebugString

### PR DESCRIPTION
**Pull request by yanboliang**
GLM toString prints out intercept, numFeatures.
For LogisticRegression and SVM model, toString also prints out numClasses, threshold.
GLM toDebugString prints out the whole weights, intercept.

### CODE REVIEW

File: mllib/src/main/scala/org/apache/spark/mllib/classification/LogisticRegression.scala [hunk 1]
```diff
@@ -163,6 +163,10 @@ class LogisticRegressionModel (
   }
 
   override protected def formatVersion: String = "1.0"
+
+  override def toString: String = {
+    super.toString + s", numClasses = $numClasses, threshold = ${threshold}"
```
Comment (srowen) [inline mllib/src/main/scala/org/apache/spark/mllib/classification/LogisticRegression.scala]: Trivial: this line uses both braces and no-braces for variables the interpolation. It's not needed in either case. The superclass `toString` is like `(weights=..., intercept=...)` so it feels a little inconsistent to continue it with `, numClasses = ...`. How about making the superclass method consistent with this?

File: mllib/src/main/scala/org/apache/spark/mllib/regression/GeneralizedLinearAlgorithm.scala [hunk 2]
```diff
@@ -76,7 +76,17 @@ abstract class GeneralizedLinearModel(val weights: Vector, val intercept: Double
...
+   * Print a summary of the model.
+   */
+  override def toString: String = {
+    this.getClass.getName + " : " + s"intercept = $intercept, numFeatures = ${weights.size}"
+  }
+
+  /**
+   * Print the full model to a string.
+   */
+  def toDebugString: String = "(weights=%s, intercept=%s)".format(weights, intercept)
```
Comment (srowen) [inline mllib/src/main/scala/org/apache/spark/mllib/regression/GeneralizedLinearAlgorithm.scala]: Why not make this use string interpolation too and use the same format?
Also, above in `toString`, why use string concatenation and interpolation? just use interpolation

Comment (srowen) [inline mllib/src/main/scala/org/apache/spark/mllib/regression/GeneralizedLinearAlgorithm.scala]: ... or really if someone wants to access the weights for debugging, won't they do so directly? this doesn't add much as a second `toString` method.

**Comment by srowen** [comment]
Now this doesn't use string interpolation at all, which is the opposite of what I suggested. I think it's better to use string interpolation, and use it consistently. I also personally do not see the value in `toDebugString` as the result is still potentially huge, and those who want access to the weights can access them directly anyway.

**Comment by yanboliang** [comment]
@srowen , sorry for the misunderstand. I have change to use string interpolation and delete ```toDebugString``` which may be useless.

**Comment by srowen** [comment]
LGTM. Will wait a short time for more comments.

**Pull request closed without merging**